### PR TITLE
SYS-3663 Implement unsigned transactions support

### PR DESCRIPTION
**Pull request by nahuseyoum**

### CODE REVIEW

File: pallets/parachain-staking/src/lib.rs [hunk 1]
```diff
@@ -1715,6 +1743,87 @@ pub mod pallet {
...
+
+                return approve_vote_validate_unsigned::<T>(
+                    &growth_voting_session,
+                    validator,
+                    eth_encoded_data.encode(),
+                    approval_signature,
+                    signature,
+                )
+            } else if let Call::reject_growth { growth_id, validator, signature } = call {
+                let growth_voting_session = Self::get_growth_voting_session(growth_id);
```
Comment (alexpArtos) [inline pallets/parachain-staking/src/lib.rs:1817]: Why don't we need to check if the growth_id.period exists, as we did in the approve branch?

File: pallets/parachain-staking/src/lib.rs [hunk 2]
```diff
@@ -2508,11 +2626,99 @@ pub mod pallet {
...
+                    growth.triggered = Some(true)
+                });
+                
+                Self::deposit_event(Event::<T>::TriggeringGrowth {
+                    growth_period: growth_id.period,                    
+                });
+            } else {
+                // We didn't get enough votes to approve this growth
+
+                let growth_creator =
```
Comment (alexpArtos) [inline pallets/parachain-staking/src/lib.rs:2686]: Is this needed at the moment? It does not seem used anywhere.

**Review by alexpArtos — APPROVED** [review]

File: pallets/parachain-staking/src/lib.rs (hunk 2 above)
Comment (nahuseyoum) [reply pallets/parachain-staking/src/lib.rs:2686]: No its not, I have another PR that cleans up unused references but will also remove it here.

File: pallets/parachain-staking/src/lib.rs (hunk 1 above)
Comment (nahuseyoum) [reply pallets/parachain-staking/src/lib.rs:1825]: Good point, added extra check.

**Pull request merged**